cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/DiagnosticsProcessor.java

```diff
@@ -102,14 +117,30 @@ private Module createModule(final ModuleConfiguration moduleConfiguration) throw
     }
 
     /**
-     * Process a query with all configured modules.
+     * Accepts a query sent by the connector. with all configured modules.
      *
      * @param query query to process
      */
     public void process(final Query query) {
-        logger.trace("Processing query {}", query);
-        for (Module module : modules) {
-            module.process(query);
+        logger.trace("Accepting query {}", query);
+        if (!queryBufferQueue.offer(query)) {
+            // the queue is full
+            logger.warn("The query buffer queueu is full. The reported query is ignored.");
+        }
+    }
+
+    private void process() {
+        while (true) {
+            try {
+                Query query = queryBufferQueue.take();
```

Comment:
This should work relatively equal to direct query handling but will it impact request rate module or any other relying on real-time query measurements?